src/vs/platform/terminal/node/windowsShellHelper.ts

```diff
@@ -28,17 +35,19 @@ const SHELL_EXECUTABLES = [
 let windowsProcessTree: typeof WindowsProcessTreeType;
 
 export class WindowsShellHelper extends Disposable implements IWindowsShellHelper {
-	private _onCheckShell: Emitter<Promise<string> | undefined> = this._register(new Emitter<Promise<string> | undefined>());
 	private _isDisposed: boolean;
 	private _currentRequest: Promise<string> | undefined;
-	private _newLineFeed: boolean = false;
-
-	private readonly _onShellNameChange = new Emitter<string>();
-	public get onShellNameChange(): Event<string> { return this._onShellNameChange.event; }
+	private _lastShellType: TerminalShellType | undefined;
+	public get lastShellType(): TerminalShellType | undefined { return this._lastShellType; }
+	private _lastShellTitle: string = '';
+	public get lastShellTitle(): string { return this._lastShellTitle; }
```

Comment:
We should remove "last" here are these are the "current" values?

Reply:
Will remove the getters, but need to keep the _lastShellType variable so I know if type has changed (to prevent flickering)